### PR TITLE
cli: make `WorkspaceCommandHelper` keep an `Arc<CommandHelper>`

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/cli_util.rs [hunk 1]
```diff
@@ -3065,7 +3068,7 @@ impl CliRunner {
         }
 
         let settings = UserSettings::from_config(config);
-        let command_helper = CommandHelper {
+        let command_helper = Rc::new(CommandHelper {
```
Comment (yuja) [inline cli/src/cli_util.rs:3082]: I think it's better to wrap inner data by `Rc` (something like `CommandHelper { data: Rc<CommandHelperData> }`) than changing function signature.

File: cli/examples/custom-backend/main.rs
Comment (yuja) [inline cli/examples/custom-backend/main.rs:1]: nit: `Arc` -> `Rc` in commit messages

**Review by yuja — APPROVED** [review]

File: cli/src/cli_util.rs [hunk 2]
```diff
@@ -613,9 +611,7 @@ impl WorkspaceCommandHelper {
             base: workspace.workspace_root().clone(),
         };
         let helper = Self {
-            string_args: command.string_args.clone(),
-            global_args: command.global_args.clone(),
-            settings,
```
Comment (yuja) [inline cli/src/cli_util.rs:615]: nit: `settings` no longer has to be cloned above

File: cli/examples/custom-backend/main.rs
Comment (martinvonz) [reply cli/examples/custom-backend/main.rs:1]: Obsolete

File: cli/src/cli_util.rs (hunk 2 above)
Comment (martinvonz) [reply cli/src/cli_util.rs:615]: Obsolete

File: cli/src/cli_util.rs [hunk 3]
```diff
@@ -573,9 +591,7 @@ impl AdvanceBranchesSettings {
 /// Provides utilities for writing a command that works on a [`Workspace`]
 /// (which most commands do).
 pub struct WorkspaceCommandHelper {
-    string_args: Vec<String>,
-    global_args: GlobalArgs,
-    settings: UserSettings,
+    command: CommandHelper,
```
Comment (yuja) [inline cli/src/cli_util.rs:594]: nit: `command.data` could be cloned instead, but I have no idea which is better overall.

File: cli/src/cli_util.rs [hunk 4]
```diff
@@ -598,26 +614,26 @@ impl WorkspaceCommandHelper {
...
-            global_args: command.global_args.clone(),
-            settings,
+            command: command.clone(),
             workspace,
             user_repo: ReadonlyUserRepo::new(repo),
-            revset_extensions: command.revset_extensions.clone(),
+            revset_extensions: command.data.revset_extensions.clone(),
             commit_summary_template_text,
-            commit_template_extensions: command.commit_template_extensions.clone(),
+            commit_template_extensions: command.data.commit_template_extensions.clone(),
```
Comment (yuja) [inline cli/src/cli_util.rs:636]: nit: `revset_extension` and `commit_template_extensions` can be removed, too.

File: cli/src/cli_util.rs (hunk 2 above)
Comment (yuja) [reply cli/src/cli_util.rs:615]: This is still valid.
```diff
@@ -614,7 +614,7 @@
         repo: Arc<ReadonlyRepo>,
         loaded_at_head: bool,
     ) -> Result<Self, CommandError> {
-        let settings = command.data.settings.clone();
+        let settings = &command.data.settings;
         let commit_summary_template_text =
             settings.config().get_string("templates.commit_summary")?;
         let revset_aliases_map =
```
(or inline it)

**Review by yuja — APPROVED** [review]

File: cli/src/cli_util.rs (hunk 2 above)
Comment (martinvonz) [reply cli/src/cli_util.rs:615]: Sorry, fixed now. I also went through and replaced a bunch of `&command.data.settings` by `command.settings()` and similar.

File: cli/src/cli_util.rs (hunk 4 above)
Comment (martinvonz) [reply cli/src/cli_util.rs]: Done

File: cli/src/cli_util.rs (hunk 3 above)
Comment (martinvonz) [reply cli/src/cli_util.rs:594]: I'm not sure either. By keeping the whole `CommandHelper` here, we can at least use its public interface, which I have started doing.

**Pull request merged**